robot-agent/packageMonitor.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+
+const constants = require('./constants');
+const { spawn, exec, execSync } = require('child_process');
+const { EventEmitter } = require('events');
+
+const { getLogger } = require('@transitive-sdk/utils');
+const log = getLogger('packageMonitor.js');
+log.setLevel('info');
+
+class PackageMonitor extends EventEmitter {
+  constructor() {
+    super();
+    this.installedPackages = {};
+  }
+  async initialize() {
+    await this.updateAllInstalledPackages();
+  }
+  async updateAllInstalledPackages() {
```

Comment:
I found that function name confusing because "update" in the agent usually means updating the installed package itself (on disk), not the information about it. Maybe `refreshAllInstalledPackages`?